atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -633,3 +495,243 @@ def run(self):
                        json.dumps(build_info, sort_keys=True, indent=4))
 
         return build_id
+
+
+class KojiImportPlugin(KojiImportBase):
+
+    key = PLUGIN_KOJI_IMPORT_PLUGIN_KEY  # type: ignore
+
+    def __init__(self, tasker, workflow,
+                 kojihub=None, url=None,
+                 verify_ssl=True, use_auth=True,
+                 koji_ssl_certs=None, koji_proxy_user=None,
+                 koji_principal=None, koji_keytab=None,
+                 blocksize=None,
+                 target=None, poll_interval=5):
+        super(KojiImportPlugin, self).__init__(tasker, workflow,
+                                               kojihub, url,
+                                               verify_ssl, use_auth,
+                                               koji_ssl_certs, koji_proxy_user,
+                                               koji_principal, koji_keytab,
+                                               blocksize,
+                                               target, poll_interval)
+
+    def set_media_types(self, extra, worker_metadatas):
+        media_types = []
+
+        # Set media_types for the base case
+        super(KojiImportPlugin, self).set_media_types(extra, worker_metadatas)
+        # Adjust media_types to include annotations
+        for platform in worker_metadatas:
+            annotations = get_worker_build_info(self.workflow,
+                                                platform).build.get_annotations()
+            if annotations.get('media-types'):
+                media_types = json.loads(annotations['media-types'])
+                break
+
+        # Append new if any, and re-sort media_types
+        existing_mts = extra['image'].get('media_types')
+        if media_types and existing_mts:
+            extra['image']['media_types'] = sorted(set(existing_mts.append(media_types)))
```

Comment:
AFAIK append/extend (it should be extend in this case) do not return the object they are called on, i.e. `some_list.extend(some_other_list) -> modifies some_list, returns None`. You could use the `+` operator instead

Not sure if this is related to the failure in unittests though

Reply:
D'OH!

Reply:
It was apparently a combination of this, plus the fact that (inexplicably) `set([].extend([]))` throws "'NoneType' object is not iterable"